core/indexer/elasticClientCommon.go

```diff
@@ -55,9 +55,16 @@ func loadResponseBody(body io.ReadCloser, dest interface{}) error {
 
 func elasticDefaultErrorResponseHandler(res *esapi.Response) error {
 	responseBody := map[string]interface{}{}
-	decodeErr := loadResponseBody(res.Body, &responseBody)
-	if decodeErr != nil {
-		return decodeErr
+	bodyBytes, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		log.Warn("cannot read elastic response body bytes", "error", err)
+		return err
+	}
+
+	notJson := json.Unmarshal(bodyBytes, &responseBody)
+	if notJson != nil {
+		log.Warn("cannot unmarshal elastic response body to map[string]interface{}", "decode error", notJson, "body response", string(bodyBytes))
```

Comment:
let's return here a wrapped error instead of keeping the wrong pattern of re-logging the same thing over and over again.

Reply:
This is a special case that can be easily treated. Totally agree with the re-logging. First, it should be a general rule, that **we don't log an error if we return it**. Special cases where you return another error than what is received can be logged in my opinion.

Here I think is safe to say something like: `received error from elastic service {error}, returning back off to caller`

Reply:
done